data/changelog/ppxlib/2023-06-19-ppxlib-0.30.0.md

```diff
@@ -0,0 +1,35 @@
+---
+title: Ppxlib 0.23.0
+date: "2023-06-19"
+tags: [ppxlib, release]
+changelog: |
+  - Clean up misleading attribute hints when declared for proper context. (#425, @ceastlund)
+
+  - Ast_pattern now has ebool, pbool helper, and a new map.(#402, @burnleydev1)
+
+  - multiple errors are now reported in `metaquot`. (#397, @burnleydev1)
+
+  - Add `Attribute.declare_with_attr_loc` (#396, @dvulakh)
+
+  - Add "ns" and "res" as reserved namespaces(#388, @davesnx)
+
+  - Make quoter `let` binding non-recursive (#401, @sim642)
+
+  - Fix failure of 'lift_map_with_context' in traverse by compile-time
+    evaluation of 'fst' and 'snd' (#390, @smuenzel)
+
+  - Driver: Bias the mapping from magic to version towards the current version,
+    as it is usually the common case and it helps when magic numbers are
+    ambiguous (such as on development versions) (#409, @shym)
+
+  - Remove unnecessary test dependencies towards base and stdio (#421, @kit-ty-kate)
+
+  - Update description to reflect that `ppxlib` contains more than a library
+    (#422, @pitag-ha)
+
+  - Add support for OCaml 5.1 (#428, @shym, @Octachron , @pitag-ha, @panglesd)
+---
+
+We're pleased to announce the release of Ppxlib 0.30.0.
+
+This release comes with support for OCaml 5.1 and brings several improvements and bug fixes.
```

Comment:
```suggestion
This release comes with support for OCaml 5.1 and brings several improvements and bug fixes.
On the side of writing PPXs, the user experience of `Ast_pattern` and `metaquot` have been
improved by adding more utility functions to the former and improving the error reporting of
the latter. On the side of using PPXs, the experience of compatibility with OCaml `trunk`
has been improved: the Driver gets biased into recognizing the accurate Parsetree version on
non-unique Parsetree magic numbers, allowing for general compatibility with both `trunk` and
stable OCaml versions at the same time. Furthermore, the release makes Ppxlib compatible
with ReScript by adding "ns" and "res" as reserved namespaces.
```

Reply:
That's perfect, I'll use this and rephrase it if needed, thanks!